src/lib/components/Form/field-mapper.js

```diff
@@ -195,8 +195,13 @@ const getFormConfig = function ({ columns, tabs = {}, getRecordAndLookups, id })
         if (!Component || (column.hideInAddGrid && id === '0')) {
             continue;
         }
+
+        const copiedColumn = { ...column };
+        if(searchParams.has('showRelation')){
+            copiedColumn.readOnly = true
+        }
         const target = tab && tabs[tab] ? tabColumns[tab] : formElements;
-        target.push({ Component, field, fieldLabel, column, otherProps });
+        target.push({ Component, field, fieldLabel, column: copiedColumn, otherProps });
```

Comment:
dont create new property, spread the object and add property along with comments.